spec/requests/missions_spec.rb

```diff
@@ -0,0 +1,177 @@
+      end
+
+      it 'sends an error message when mission is regulated and duration is not a multiple of 1.5 hours' do
+        invalid_mission_params = invalid_attributes
+
+        post missions_path, params: { mission: invalid_mission_params }
+        follow_redirect!
+
+        expect(CGI.unescapeHTML(response.body))
+          .to include(I18n.t('activerecord.errors.models.mission.attributes.duration.multiple'))
+      end
+    end
+  end
+
+  describe 'GET edit' do
+    before { sign_in create :member, :super_admin }
+
+    context 'when the mission is not regulated' do
+      it 'rends the standards partials' do
+        mission = create :mission
+
+        get edit_mission_path(mission.id)
+
+        expect(response).to render_template(partial: 'missions/_enrollments_form')
+      end
+    end
+
+    context 'when the mission is regulated' do
+      it 'rends the regulated partials' do
+        mission = create :mission, genre: 'regulated'
+
+        get edit_mission_path(mission.id)
+
+        expect(response).to render_template(partial: 'missions/_enrollments_with_time_slots_form')
+      end
+    end
+  end
+
+  describe 'PUT' do
+    subject(:put_mission) { put mission_path(mission.id), params: { mission: mission_params } }
+
+    let(:mission) { create :mission }
+    let(:mission_params) {  { name: 'updated_mission' } }
+
+    before { sign_in create :member, :super_admin }
+
+    it 'updates the mission' do
+      put_mission
+
+      expect(mission.reload.name).to eq 'updated_mission'
+    end
+
+    context 'when the mission is not regulated' do
+      let(:other_member) { create :member }
+      let(:enrollment_params) do
+        { member_id: other_member.id, start_time: mission.start_date, end_time: mission.due_date }
+      end
+      let(:mission_params) do
+        { name: 'updated_mission', enrollments_attributes: { '1234': enrollment_params } }
+      end
+
+      it 'adds member enrollment' do
+        put mission_path(mission.id), params: { mission: mission_params }
+
+        enrollment_params.each do |key, value|
+          expect(mission.enrollments.first[key]).to eq value
+        end
+      end
+    end
+
+    context 'when the mission is regulated' do
+      let(:other_member) { create :member }
+
+      let(:enrollment_expected_params) do
+        { member_id: other_member.id, start_time: mission.start_date, end_time: mission.start_date + 3.hours }
+      end
+
+      let(:enrollment_params) do
+        { member_id: other_member.id, start_time: [mission.start_date, mission.start_date + 90.minutes] }
+      end
+
+      let(:mission_params) do
+        { name: 'updated_mission', enrollments_attributes: { '1234': enrollment_params } }
+      end
+
+      it 'adds member enrollment' do
+        put mission_path(mission.id), params: { mission: mission_params }
+
+        enrollment_expected_params.each do |key, value|
+          expect(mission.enrollments.first[key]).to eq value
+        end
+      end
+    end
```

Comment:
```ruby
context 'when enrollment params are given ' do
  # put here examples for a 'normal' mission (=not regulated)
  ...

  context 'with a :regulated mission do
    # put here examples for a :regulated mission
  end
end
```

I think writing it like this would frame the 'enrollment' update better. What do you think?

Reply:
i agree with you